apps/guardian/src-tauri/src/main.rs

```diff
@@ -181,17 +196,40 @@ struct ActiveGuardian {
 #[derive(Default)]
 struct ActiveGuardianState(Mutex<Option<ActiveGuardian>>);
 
-/// The collective state kept for all guardians
+/// The data registered in the election contract
+#[derive(Default)]
+struct ContractData {
+    /// The guardians state registered in the election contract
+    guardians:       contract::GuardiansState,
+    /// The encrypted tally registered in the contract
+    encrypted_tally: Option<EncryptedTally>,
+}
```

Comment:
Yes this might seem weird, but to keep the user flows simple the validation of what has been shared in the contract is done when that specific thing is used, e.g. public keys are used parsed when encrypted shares are created. If this cannot be done, an alternate route is proposed to the user, which is to file a complaint with all the invalid key registrations.